src/allmydata/test/test_storage_http.py

```diff
@@ -474,6 +488,20 @@ def write(offset, length):
             )
             self.assertEqual(downloaded, expected_data[offset : offset + length])
 
+    def test_write_with_wrong_upload_key(self):
+        """A write with the wrong upload key fails."""
```

Comment:
maybe worth expanding "wrong key" to something like "a key not equal to the original upload key"?